File: Web/src/api/stock/stoAdjustItem.ts
import {useBaseApi} from '/@/api/base';

// 调整单明细表接口服务
export const useStoAdjustItemApi = () => {
	const baseApi = useBaseApi("stoAdjustItem");
	return {
		// 分页查询调整单明细表
		page: baseApi.page,
		// 查看调整单明细表详细
		detail: baseApi.detail,
		// 新增调整单明细表
		add: baseApi.add,
		// 更新调整单明细表
		update: baseApi.update,
		// 删除调整单明细表
		delete: baseApi.delete,
		// 批量删除调整单明细表
		batchDelete: baseApi.batchDelete,
		// 导出调整单明细表数据
		exportData: baseApi.exportData,
		// 导入调整单明细表数据
		importData: baseApi.importData,
		// 下载调整单明细表数据导入模板
		downloadTemplate: baseApi.downloadTemplate,
	}
}

// 调整单明细表实体
export interface StoAdjustItem {
	// 主键Id
	id: number;
	// 调整单ID
	adjustId?: number;
	// 物料ID
	materialId?: number;
	// 物料编码
	materialCode?: string;
	// 物料名称
	materialName?: string;
	// 规格型号
	spec: string;
	// 单位
	unit?: string;
	// 调整数量(正数增加,负数减少)
	adjustQuantity?: number;
	// 成本单价
	costPrice?: number;
	// 调整金额
	adjustAmount?: number;
	// 库位编码
	locationCode: string;
	// 批号
	batchNo: string;
	// 排序号
	sortOrder?: number;
	// 备注
	remark: string;
	// 创建时间
	createTime: string;
	// 更新时间
	updateTime: string;
	// 创建者Id
	createUserId: number;
	// 创建者姓名
	createUserName: string;
	// 修改者Id
	updateUserId: number;
	// 修改者姓名
	updateUserName: string;
}

File: CzhERP.Application/Entity/StoAdjustItem.cs
namespace CzhERP.Application.Entity;

[Tenant("1300000000001")]
[SugarTable("Sto_AdjustItem", "调整单明细表")]
public partial class StoAdjustItem : EntityBase
{
    [Required]
    [SugarColumn(ColumnName = "AdjustId", ColumnDescription = "调整单ID")]
    public virtual long AdjustId { get; set; }

    [Required]
    [SugarColumn(ColumnName = "MaterialId", ColumnDescription = "物料ID")]
    public virtual long MaterialId { get; set; }

    [Required]
    [SugarColumn(ColumnName = "MaterialCode", ColumnDescription = "物料编码", Length = 50)]
    public virtual string MaterialCode { get; set; }

    [Required]
    [SugarColumn(ColumnName = "MaterialName", ColumnDescription = "物料名称", Length = 100)]
    public virtual string MaterialName { get; set; }

    [SugarColumn(ColumnName = "Spec", ColumnDescription = "规格型号", Length = 100)]
    public virtual string? Spec { get; set; }

    [Required]
    [SugarColumn(ColumnName = "Unit", ColumnDescription = "单位", Length = 20, DefaultValue = "个")]
    public virtual string Unit { get; set; } = "个";

    [Required]
    [SugarColumn(ColumnName = "AdjustQuantity", ColumnDescription = "调整数量(正数增加,负数减少)")]
    public virtual decimal AdjustQuantity { get; set; }

    [SugarColumn(ColumnName = "CostPrice", ColumnDescription = "成本单价", DefaultValue = "0")]
    public virtual decimal CostPrice { get; set; } = 0;

    [SugarColumn(ColumnName = "AdjustAmount", ColumnDescription = "调整金额", DefaultValue = "0")]
    public virtual decimal AdjustAmount { get; set; } = 0;

    [SugarColumn(ColumnName = "LocationCode", ColumnDescription = "库位编码", Length = 50)]
    public virtual string? LocationCode { get; set; }

    [SugarColumn(ColumnName = "BatchNo", ColumnDescription = "批号", Length = 50)]
    public virtual string? BatchNo { get; set; }

    [SugarColumn(ColumnName = "SortOrder", ColumnDescription = "排序号", DefaultValue = "0")]
    public virtual int SortOrder { get; set; } = 0;

    [SugarColumn(ColumnName = "Remark", ColumnDescription = "备注", Length = 200)]
    public virtual string? Remark { get; set; }
}
